### DISCUSSION TITLE
[@types/hapi__hapi] 20.0.12 does not work with joi 17.6.3

### DISCUSSION THREAD

**Discussion by radoslavirha** [DefinitelyTyped/DefinitelyTyped#62701] [Issues with a @types package]
Hi,

I upgraded joi to latest version, but Hapi is not happy with it.

`server.validator(Joi)`

TS throws `The types of 'any().alter' are incompatible between these types`

Thanks

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
@Radik24 As I see it works now. Can you recheck?

**Reply by radoslavirha** [reply]
Hi, it's still not working for me when I upgrade Joi. But i decided to bump types first, fix some types and try with joi later.

There is a huge improve in typings in 20.0.13, but I still struggle with server.app typing (ServerApplicationState interface). More specifically now I cannot correctly set types for request.server.app in handlers. Before I did lots of types to get it working. 

Do you think is it possible to create it as a composite type too?

**Reply by lenovouser** [reply]
@damusix maybe this is something you can do 👍🏻
